numpydoc/docscrape.py

```diff
@@ -252,48 +272,62 @@ def _parse_see_also(self, content):
 
         def parse_item_name(text):
             """Match ':role:`name`' or 'name'"""
-            m = self._name_rgx.match(text)
-            if m:
-                g = m.groups()
-                if g[1] is None:
-                    return g[3], None
-                else:
-                    return g[2], g[1]
-            raise ParseError("%s is not a item name" % text)
+            m = self._func_rgx.match(text)
+            if not m:
+                raise ParseError("%s is not a item name" % text)
+            role = m.groupdict().get('role')
+            if role:
+                name = m.group('name')
+            else:
+                name = m.group('name2')
+            return name, role, m
 
         def push_item(name, rest):
             if not name:
                 return
-            name, role = parse_item_name(name)
+            name, role, m2 = parse_item_name(name)
             items.append((name, list(rest), role))
             del rest[:]
 
-        current_func = None
         rest = []
 
         for line in content:
             if not line.strip():
                 continue
 
-            m = self._name_rgx.match(line)
-            if m and line[m.end():].strip().startswith(':'):
-                push_item(current_func, rest)
-                current_func, line = line[:m.end()], line[m.end():]
-                rest = [line.split(':', 1)[1].strip()]
-                if not rest[0]:
-                    rest = []
-            elif not line.startswith(' '):
-                push_item(current_func, rest)
-                current_func = None
-                if ',' in line:
-                    for func in line.split(','):
-                        if func.strip():
-                            push_item(func, [])
-                elif line.strip():
-                    current_func = line
-            elif current_func is not None:
+            ml = self._line_rgx.match(line)
+            description = None
+            if ml:
+                if 'description' in ml.groupdict():
+                    description = ml.groupdict().get('desc')
+            if not description and line.startswith(' '):
                 rest.append(line.strip())
-        push_item(current_func, rest)
+            elif ml:
+                funcs = []
+                text = ml.group('allfuncs')
+                while True:
+                    if not text.strip():
+                        break
+                    name, role, m2 = parse_item_name(text)
```

Comment:
It would be cleaner to have `parse_item_name` return the `end` than the match. especially if you're going to call it the inscrutable `m2`

Reply:
Done.